tests/FixerFactoryTest.php

```diff
@@ -250,6 +251,82 @@ public function testNoDoubleConflictReporting()
         );
     }
 
+    public function testSetWhitespacesConfig()
+    {
+        $factory = new FixerFactory();
+        $config = new WhitespacesFixerConfig();
+
+        $fixer = $this->prophesize('PhpCsFixer\Fixer\WhitespacesAwareFixerInterface');
+        $fixer->getName()->willReturn('foo');
+        $fixer->setWhitespacesConfig($config)->shouldBeCalled();
+
+        $factory->registerFixer($fixer->reveal(), false);
+
+        $factory->setWhitespacesConfig($config);
+    }
+
+    public function testRegisterFixerInvalidName()
+    {
+        $this->setExpectedException('UnexpectedValueException', 'Fixer named "0" has invalid name.');
```

Comment:
The exception expectation should be set as late as possible before executing the code the code that is supposed to throw it. This will make the test fail if the exception is thrown before the expectation is set. I know it's very unlikely here but I think it's a good practice that we should always follow :)